### DISCUSSION TITLE
20230819周会纪要

### DISCUSSION THREAD

**Discussion by Mulavar** [arana-db/arana#740] [Weekly meeting]
## 1. 本周进展

| 功能点 | 参与人员 | 本周工作进展 | 备注 |
| --- | --- | --- | --- |
| [Test] add unit test for pkg/executor and pkg/merge <br/> [Test] add unit test for /arana/pkg/sequence/... package. | @dongjh |  | [#709](https://github.com/arana-db/arana/issues/709) https://github.com/arana-db/arana/issues/731|
| 1、[TX] support transaction XA  @chunshao  <br> ~~2、[XA] build auto clean transaction log ability @pangx~~ <br> ~~3、[XA] build transaction log ability @liy~~ <br> 4、[XA] build XA transaction back-and-forth decision-making  @haohf <br> 5、[tx] add deadlock for arana transaction. @liuyc  | @chunshao <br> @liuyc <br> @haohf <br> @liy（finish）<br> @pangx（finish） |  | [https://github.com/arana-db/arana/issues/324](https://github.com/arana-db/arana/issues/324)  <br> [https://github.com/arana-db/arana/issues/555](https://github.com/arana-db/arana/issues/555) <br> [https://github.com/arana-db/arana/issues/602](https://github.com/arana-db/arana/issues/602) <br> [https://github.com/arana-db/arana/issues/607](https://github.com/arana-db/arana/issues/607) |
| [Feature] Support SQL join | @sudu @huangwk |  | [https://github.com/arana-db/arana/issues/600](https://github.com/arana-db/arana/issues/600) <br> [https://github.com/arana-db/arana/issues/622](https://github.com/arana-db/arana/issues/622) |
|  | @wangrui  | 新issue |  |
| [Test] add unit test for /arana/pkg/mysql/... package.  | @pangx | | https://github.com/arana-db/arana/issues/711 |
| [Prometheus] <br> [Test] add unit test for /arana/pkg/registry/... package. <br> [Test] add unit test for /arana/pkg/proto/... package.  | @gongna |   | https://github.com/arana-db/arana/issues/714 <br> https://github.com/arana-db/arana/issues/712
| [Bug] fix show tables xxx statement error.  | @zhangxy |  | [https://github.com/arana-db/arana/issues/677](https://github.com/arana-db/arana/issues/677) |
| [Test] add unit test for /arana/pkg/mysql package  | @weideman |  | https://github.com/arana-db/arana/issues/710 |
| [Test] add unit test for /arana/pkg/dataset <br> [Test] add unit test for /arana/pkg/boot. <br> [Pendding] MySQL  window 函数实现 | @yangwei |  | https://github.com/arana-db/arana/issues/708 <br> https://github.com/arana-db/arana/issues/706 <br> [Pending][https://github.com/arana-db/arana/issues/667](https://github.com/arana-db/arana/issues/667) |
| [SQL] Support MySQL Prepare SQL | @maronghe | 正在调研 https://github.com/arana-db/arana/discussions/700 | [https://github.com/arana-db/arana/issues/591](https://github.com/arana-db/arana/issues/591) |
|  | @maorui | 新issue |  |
| [SQL] Support show table rules from xxx | @zhangbinbin |  | [https://github.com/arana-db/arana/issues/628](https://github.com/arana-db/arana/issues/628) |
| [Bug]Optimize show open tables xxx statment. | @huxiaohan | | [https://github.com/arana-db/arana/issues/676](https://github.com/arana-db/arana/issues/676) |
| [Test] integrated ChatGPT for unit test code. | @pengjiaxin | | https://github.com/arana-db/arana/issues/748 |
| [kernel] run arana at the envoy. | | | [https://github.com/arana-db/arana/issues/649](https://github.com/arana-db/arana/issues/649) |

## 2. 下周工作
| 功能点 | 参与人员 | 备注 |
| --- | --- | --- | 
| [Test] add unit test for pkg/executor and pkg/merge <br/> [Test] add unit test for /arana/pkg/sequence/... package. | @dongjh  | [#709](https://github.com/arana-db/arana/issues/709) https://github.com/arana-db/arana/issues/731|
| 1、[TX] support transaction XA  @chunshao  <br> ~~2、[XA] build auto clean transaction log ability @pangx~~ <br> ~~3、[XA] build transaction log ability @liy~~ <br> 4、[XA] build XA transaction back-and-forth decision-making  @haohf <br> 5、[tx] add deadlock for arana transaction. @liuyc  | @chunshao <br> @liuyc <br> @haohf <br> @liy（finish）<br> @pangx（finish） | [https://github.com/arana-db/arana/issues/324](https://github.com/arana-db/arana/issues/324)  <br> [https://github.com/arana-db/arana/issues/555](https://github.com/arana-db/arana/issues/555) <br> [https://github.com/arana-db/arana/issues/602](https://github.com/arana-db/arana/issues/602) <br> [https://github.com/arana-db/arana/issues/607](https://github.com/arana-db/arana/issues/607) |
| [Feature] Support SQL join | @sudu @huangwk  | [https://github.com/arana-db/arana/issues/600](https://github.com/arana-db/arana/issues/600) <br> [https://github.com/arana-db/arana/issues/622](https://github.com/arana-db/arana/issues/622) |
|  | @wangrui  |  |
| [Test] add unit test for /arana/pkg/mysql/... package.  | @pangx | https://github.com/arana-db/arana/issues/711 |
| [Prometheus] <br> [Test] add unit test for /arana/pkg/registry/... package. <br> [Test] add unit test for /arana/pkg/proto/... package.  | @gongna  | https://github.com/arana-db/arana/issues/714 <br> https://github.com/arana-db/arana/issues/712
| [Bug] fix show tables xxx statement error.  | @zhangxy  | [https://github.com/arana-db/arana/issues/677](https://github.com/arana-db/arana/issues/677) |
| [Test] add unit test for /arana/pkg/mysql package  | @weideman | https://github.com/arana-db/arana/issues/710 |
| [Test] add unit test for /arana/pkg/dataset <br> [Test] add unit test for /arana/pkg/boot. <br> [Pendding] MySQL  window 函数实现 | @yangwei | https://github.com/arana-db/arana/issues/708 <br> https://github.com/arana-db/arana/issues/706 <br> [Pending][https://github.com/arana-db/arana/issues/667](https://github.com/arana-db/arana/issues/667) |
| [SQL] Support MySQL Prepare SQL | @maronghe | https://github.com/arana-db/arana/discussions/700 | [https://github.com/arana-db/arana/issues/591](https://github.com/arana-db/arana/issues/591) |
|  | @maorui | |
| [SQL] Support show table rules from xxx | @zhangbinbin | [https://github.com/arana-db/arana/issues/628](https://github.com/arana-db/arana/issues/628) |
| [Bug]Optimize show open tables xxx statment. | @huxiaohan | [https://github.com/arana-db/arana/issues/676](https://github.com/arana-db/arana/issues/676) |
| [Test] integrated ChatGPT for unit test code. | @pengjiaxin | | https://github.com/arana-db/arana/issues/748 |
| [kernel] run arana at the envoy. | | [https://github.com/arana-db/arana/issues/649](https://github.com/arana-db/arana/issues/649) |

### 0.2.0 规划

[https://arana.yuque.com/staff-guqen5/howkif/zr5od2](https://arana.yuque.com/staff-guqen5/howkif/zr5od2)

**Comment by dongzl** [comment]
1、梳理 XA 实现方案，最近会拉小伙伴对一下实现；
2、王瑞参与事务工作；
3、庞兴演示 ChatGPT 生成 UT 代码；
4、张心雨、张斌斌
5、ChapGPT 生成单测；
6、ChapGPT 工具集成；
7、泽仁，OLAP 数据库、
